File: src/leetcode/python/BaseballGame_682.py
class BaseballGame_682(object):
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        stk = []
        for op in ops:
            if op == "+":
                stk.append(stk[-1] + stk[-2])
            elif op == "D":
                stk.append(stk[-1] * 2)
            elif op == "C":
                stk.pop()
            else:
                stk.append(int(op))
        return sum(stk)

File: src/leetcode/java/BaseballGame_682.java
package leetcode.java;

public class BaseballGame_682 {
    public int calPoints(String[] ops) {
        int[] points = new int[ops.length];
        int k = 0;
        for (int i = 0; i < ops.length; i++) {
            if (ops[i].equals("+")) {
                points[k] = points[k - 1] + points[k - 2];
                k++;
            } else if (ops[i].equals("D")) {
                points[k] = 2 * points[k - 1];
                k++;
            } else if (ops[i].equals("C")) {
                points[--k] = 0;
            } else {
                points[k++] = Integer.valueOf(ops[i]);
            }
        }
        int sum = 0;
        for (int i = 0; i < k; i++) sum += points[i];
        return sum;
    }
}
